### ISSUE TITLE
Masking of torch.long tensors cannot be compiled through llvm-cpu backend

### ISSUE THREAD

**Issue by MaxS1996** [iree-org/iree#20550]
### What happened?

Compiling a workload from PyTorch, which applies a mask to a tensor that has torch.long as dtype fails when going through the llvm-cpu backend:

```python
import torch
import torch.nn as nn

class MaskingModule(nn.Module):
    
    def __init__(self, dtype_data=torch.float, dtype_mask=torch.long):
        super(MaskingModule, self).__init__()
        self.dtype_data = dtype_data
        self.dtype_mask = dtype_mask

    def forward(self, x):
        tensor_a = torch.zeros_like(x, dtype=self.dtype_data)
        tensor_b = torch.ones_like(x, dtype=self.dtype_data) * 2

        mask = torch.rand_like(x) * 10
        mask = mask < 5
        mask = mask.to(self.dtype_mask)

        tensor_a[mask] = tensor_b[mask]
        return tensor_a

# Example usage
shape = (2, 100, 3) # or (2, 10, 3)
dtype_data = torch.long
dtype_mask = torch.long # only long currently supported by IREE, see issue #20549
x = torch.rand(shape)
example_args = (x,)
model_torch = MaskingModule(dtype_data, dtype_mask)
out = model_torch(*example_args)

import iree.turbine.aot as aot

export_output = aot.export(model_torch, *example_args)
binary = export_output.compile(
    target_backends=["llvm-cpu"], 
    save_to="test.vmfb"
)

```

The error that occurs seems to depend on the shape of the input tensor.
Smaller tensors result in: **error: 'linalg.fill' op write affecting operations on global resources are restricted to workgroup distributed contexts.; error: 'func.func' op failed on workgroup distribution verification**

Larger tensors cause: **error: 'func.func' op exceeded stack allocation limit of 32768 bytes for function. Got 90048 bytes**

Both errors do not occur for the cuda backend, which I used to compare the behaviour to.

### Steps to reproduce your issue

1. Define a model that uses masking with dtype=torch.long.
2. Compile the model using the llvm-cpu backend.
3. Depending on the tensor shape, the compile step will fail with one of the described error messages

Observed Behavior:

* Compilation fails when using dtype=torch.long for the mask.
* Compiling with the cuda backend works.
* Tensors with dtype=int32 are not affected.

### Version information

3.3.0 for compiler, turbine and runtime

**Comment by MaxS1996** [comment]
might be related to #20549 and #20541
